### ISSUE TITLE
[rush] After "rush init", build fails with "Unexpected cobuild lock provider"

### ISSUE THREAD

**Issue by octogonz** [microsoft/rushstack#4307]
## Summary

The `rush init` template generates **cobuild.json** but this file is not usable without a Cobuild lock provider plugin.

## Repro steps

1. `rush init` in a new empty repo, or `rush init --overwrite-existing` in an existing repo
2. Commit
3. `rush update`

It prints this error:

```
ERROR: Unexpected cobuild lock provider: redis
```

But cobuilds are disabled:

**common/config/rush/cobuild.json**
```js
{
  "$schema": "https://developer.microsoft.com/json-schemas/rush/v5/cobuild.schema.json",

  /**
   * (Required) EXPERIMENTAL - Set this to true to enable the cobuild feature.
   * RUSH_COBUILD_CONTEXT_ID should always be specified as an environment variable with an non-empty string,
   * otherwise the cobuild feature will be disabled.
   */
  "cobuildFeatureEnabled": false,

  /**
   * (Required) Choose where cobuild lock will be acquired.
   *
   * The lock provider is registered by the rush plugins.
   * For example, @rushstack/rush-redis-cobuild-plugin registers the "redis" lock provider.
   */
  "cobuildLockProvider": "redis"
}
```

And I can't remove the `"cobuildLockProvider"` line because it is REQUIRED.

  **Expected result:** 

The `"cobuildLockProvider"` setting should not be validated unless `cobuildFeatureEnabled=true`

## Details
This is a small regression from the new Cobuild feature. It only affects people who run `rush init`.

There is a trivial workaround: delete **cobuild.json**.  But the experience should be better.

## Standard questions

Please answer these questions to help us investigate your issue more quickly:

| Question | Answer |
| -------- | -------- |
| `@microsoft/rush` globally installed version? |   5.104.0|
| `rushVersion` from rush.json? | l 5.104.0 |
| `useWorkspaces` from rush.json? |  true|
| Operating system? | Windows |
| Node.js version (`node -v`)? |  16.20.2|

@chengcyber

**Comment by aruntj** [comment]
I'm facing the same issue. Deleting cobuild.json worked for me.

**Comment by octogonz** [comment]
Here's a PR: https://github.com/microsoft/rushstack/pull/4309

**Comment by octogonz** [comment]
Fixed in Rush 5.104.1
